doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,860 @@
+                    description "Disable ARS forwarding on matching packets";
+                    type boolean;
+                    default false;
+                }
+            }
+         }
+    }
+```
+
+#### Config DB Enhancements  
+
+```
+; New container ARS_PROFILE_TABLE
+; ARS global configuration
+
+key                     = ARS_PROFILE|profile_name
+
+;field                  = value
+
+algorithm               = "ewma"        ;Path quality calculation algorithm
+max_flows               = uint32        ;Maximum number of flows that can be maintained for ARS
+sampling_interval       = uint32        ;Sampling interval in microseconds
+past_load_min_value     = uint16        ;Minimum value of Past load range.
+past_load_max_value     = uint16        ;Maximum value of Past load range.
+past_load_weight        = uint16        ;Weight of the past load
+future_load_min_value   = uint16        ;Minimum value of Future load range.
+future_load_max_value   = uint16        ;Maximum value of Future load range.
+future_load_weight      = uint16        ;Weight of the future load
+current_load_min_value  = uint16        ;Minimum value of Current load range.
+current_load_max_value  = uint16        ;Maximum value of Current load range.
+ipv4_enable             = boolean       ;Whether ARS is enabled over IPv4 packets
+ipv6_enable             = boolean       ;Whether ARS is enabled over IPv6 packets
+
+
+Configuration exmaple:
+
+"ARS_PROFILE": {
+    "default": {
+        "algorithm": "ewma",
+        "max_flows" : 512,
+        "sampling_interval": 10,
+        "past_load_min_value" : 0,
+        "past_load_max_value" : 100,
+        "past_load_weight": 1,
+        "future_load_min_value" : 0,
+        "future_load_max_value" : 1000,
+        "future_load_weight": 5.
+        "ipv4_enable" : "true",
+        "ipv6_enable" : "true"
+    }
+}
+```
+
+```
+; New table ARS_INTERFACE_TABLE
+; ARS interfaces configuration
+
+key                      = ARS_INTERFACE|if_name          ;ifname is the name of the ARS-enabled interface
+
+Configuration exmaple:
+
+"ARS_INTERFACE": {
+    "flowlet-based | Ethernet0" : {},
+    "flowlet-based | Ethernet8" : {}
+}
+```
+
+```
+; New table ARS_NEXTHOP_GROUP_TABLE
+; Nexhop groups enabled for ARS
+
+key                      = ARS_NEXTHOP_GROUP|ip_prefix|vrf_name     ;Route prefix identifing nexhop-group 
+
+;field                   = value
+
+profile_name             = string                                   ;ARS profile Name
+assign_mode              = "per_flowlet_quality" / "per_packet"     ;member selection assignment mode
+flowlet_idle_time        = uint16                                   ;idle time for decting flowlet in macro flow. Relevant only for assign_mode=pre_flowlet
+max_flows                = uint16                                   ;Max number of flows supported for ARS
+primary_path_threshold   = uint16                                   ;Quality threshold for primary path 

```

Comment:
Primary/Alternate is not supported in TL S/W and AFAIK there is no special support in the device for these. If really needed, we need to manage in s/w.